src/Stryker.Core/Stryker.Core.UnitTest/TestRunners/VsTestMockingHelper.cs

```diff
@@ -130,19 +130,24 @@ protected SourceProjectInfo BuildSourceProjectInfo(IEnumerable<Mutant> mutants =
 
     protected IReadOnlyList<Microsoft.VisualStudio.TestPlatform.ObjectModel.TestCase> TestCases { get; }
 
-    private static void DiscoverTests(ITestDiscoveryEventsHandler discoveryEventsHandler, IReadOnlyCollection<Microsoft.VisualStudio.TestPlatform.ObjectModel.TestCase> tests, bool aborted) =>
-        Task.Run(() => discoveryEventsHandler.HandleDiscoveredTests(tests)).
+    private static Task DiscoverTests(ITestDiscoveryEventsHandler discoveryEventsHandler, IReadOnlyCollection<Microsoft.VisualStudio.TestPlatform.ObjectModel.TestCase> tests, bool aborted) =>
+         Task.Run(() => discoveryEventsHandler.HandleDiscoveredTests(tests)).
             ContinueWith((_, u) => discoveryEventsHandler.HandleDiscoveryComplete((int)u, null, aborted), tests.Count);
 
     protected Microsoft.VisualStudio.TestPlatform.ObjectModel.TestCase BuildCase(string name, TestFrameworks framework = TestFrameworks.xUnit, string displayName = null)
         => new(name, framework == TestFrameworks.xUnit ? _xUnitUri : _NUnitUri, _testAssemblyPath) { Id = new Guid(), DisplayName = displayName ?? name };
 
     private Microsoft.VisualStudio.TestPlatform.ObjectModel.TestCase FindOrBuildCase(string testResultId) => TestCases.FirstOrDefault(t => t.FullyQualifiedName == testResultId) ?? BuildCase(testResultId);
 
-    private static void MockTestRun(ITestRunEventsHandler testRunEvents, IReadOnlyList<TestResult> testResults,
+    private static Task MockTestRun(ITestRunEventsHandler testRunEvents, IReadOnlyList<TestResult> testResults,
         Microsoft.VisualStudio.TestPlatform.ObjectModel.TestCase timeOutTest = null) =>
         Task.Run(() =>
         {
+            if (testResults.Count == 0)
```

Comment:
Just thinking out loud: from a design perspective, this feels like a bit too much of white-box testing - I would expect this to be a transparent implementation detail in the code itself.

Reply:
I do not think this is white box testing, in the sense that the test verifies what happens inside the _sut_.
Here it is about better faking for VsTest: if you launch VsTest with an empty test assembly, you will not get any event at all. While I suspect this is a bug on VsTest part, we still need to handle this gracefully

Don't you think?

Reply:
Yeah fair, I didn't look closely enough. Okay then!